code/modules/power/apc.dm

```diff
@@ -1256,6 +1256,16 @@
 			charging = 0
 			chargecount = 0
 
+		//if the cell is fully charged but there is too much excess power, the APC will produce a tesla shock
+		if(excess > (cell.maxcharge * 100))
+			if(cell.charge != cell.maxcharge)
+				return
+			if(cell_type == /obj/item/stock_parts/cell/high/slime)
+				return
+			if(prob(50))
+				playsound(src, 'sound/magic/lightningshock.ogg', 100, 1, extrarange = 5)
```

Comment:
TRUE

Reply:
Is the return two lines above it fine?

Reply:
Never mind I added that check to the initial test case.